packages/thrift-server-core/src/main/Int64.ts

```diff
@@ -258,32 +261,45 @@ export class Int64 implements IInt64 {
     public static fromDecimalString(text: string): Int64 {
         const negative: boolean = text.charAt(0) === '-'
 
-        if (text.length < (negative ? 17 : 16)) {
-            // The magnitude is smaller than 2^53.
-            return new Int64(+text)
-        } else if (text.length > (negative ? 20 : 19)) {
+        // Short numbers are fast.
+        const numVal = parseInt(text, 10)
+        if (Number.isNaN(numVal)) {
+            // Not a numeric string. Return zero.
+            return new Int64(0)
+        }
+        if (Number.isSafeInteger(numVal)) {
+            // Use the number.
+            return new Int64(numVal)
+        }
+
+        // Extract just the number part.
+        const matches = text.match(DEC_REGEX)
+        const numericPart = (matches && matches[1]) || ''
```

Comment:
I was trying to avoid the regex for performance reasons.

I could leave just the short number change above to check for NaN and leave out the following changes without the regex if performance trumps some of the odd cases that start with a number but are not all number.